app/src/main/java/com/android/wardrobeManager/ui/closet/ClosetActivity.java

```diff
@@ -53,13 +83,59 @@ public void onBackPressed() {
         startActivity(new Intent(ClosetActivity.this, StartActivity.class));
     }
 
+    public void openMenu(View view) {
+        if (view.getId() == findViewById(R.id.closet_menu_button).getId()) {
+            if (!closetMenu.fragmentIsOpen() && !nonClickableShade.fragmentIsOpen() && !clickableShade.fragmentIsOpen()) {
+                menuIsOpen = true;
+                fragmentTransaction = fragmentManager.beginTransaction();
+                fragmentTransaction.setCustomAnimations(R.anim.fade_in, R.anim.fade_out);
+                fragmentTransaction.add(R.id.shade_layout_holder, clickableShade).commit();
+
+                fragmentTransaction = fragmentManager.beginTransaction();
+                fragmentTransaction.setCustomAnimations(R.anim.fade_in, R.anim.fade_out);
+                fragmentTransaction.add(R.id.shade_layout_holder, nonClickableShade).commit();
+
+                fragmentTransaction = fragmentManager.beginTransaction();
+                fragmentTransaction.setCustomAnimations(R.anim.slide_from_right, R.anim.fade_out);
+                fragmentTransaction.add(R.id.menu_layout_holder, closetMenu);
+                fragmentTransaction.commit();
+
+                findViewById(R.id.closet_menu_button).setVisibility(View.INVISIBLE);
+            }
+        }
+    }
+
     @Override
     public void onClothingItemClick(ClothingItem item, ImageButton clothingImageView) {
-        Intent intent = new Intent(ClosetActivity.this, AddItemActivity.class);
-        Bundle bundle = new Bundle();
-        bundle.putParcelable("clothingItem", item);
-        intent.putExtras(bundle);
-        ActivityOptions options = ActivityOptions.makeSceneTransitionAnimation(ClosetActivity.this, clothingImageView, "add_item_viewpager_transition");
-        startActivity(intent, options.toBundle());
+        if (!menuIsOpen)  {
+            Intent intent = new Intent(ClosetActivity.this, AddItemActivity.class);
+            Bundle bundle = new Bundle();
+            bundle.putParcelable("clothingItem", item);
+            intent.putExtras(bundle);
+            ActivityOptions options = ActivityOptions.makeSceneTransitionAnimation(ClosetActivity.this, clothingImageView, "add_item_viewpager_transition");
+            startActivity(intent, options.toBundle());
+        }
     }
+
+    public void shadeClicked(View view) {
+        if (menuIsOpen) {
+            menuIsOpen = false;
+            fragmentTransaction = fragmentManager.beginTransaction();
+            fragmentTransaction.setCustomAnimations(R.anim.fade_in, R.anim.fade_out);
+            fragmentTransaction.remove(clickableShade).commit();
+
+            fragmentTransaction = fragmentManager.beginTransaction();
+            fragmentTransaction.setCustomAnimations(R.anim.fade_in, R.anim.fade_out);
+            fragmentTransaction.remove(nonClickableShade).commit();
```

Comment:
Could you explain what the point of adding two different fragments is when they are exactly the same class? I just don't really see when the difference is and what the point is of having two of them.

Reply:
The two shade fragments are the same aside from the fact that one is able to be clicked to bring the user from the menu and back to the closet. If I only had one fragment, then the user would exit the menu when clicking in the menu area. So I made one fragment in the menu area that is not clickable to stop this and an adjacent shade fragment that brings the user back to the closet.